src/client/tile.cpp

```diff
@@ -1183,55 +1176,76 @@ void blitBaseImage(video::IImage* &src, video::IImage* &dst)
 	blit_with_alpha(src, dst, pos_from, pos_to, dim_dst);
 }
 
+#define CHECK_BASEIMG() \
+	do { \
+		if (!baseimg) { \
+			errorstream << "generateImagePart(): baseimg == NULL" \
+					<< " for part_of_name=\"" << part_of_name \
+					<< "\", cancelling." << std::endl; \
+			return false; \
+		} \
+	} while(0)
+
+#define COMPLAIN_INVALID(description) \
+	do { \
+		errorstream << "generateImagePart(): invalid " << (description) \
+			<< " for part_of_name=\"" << part_of_name \
+			<< "\", cancelling." << std::endl; \
+		return false; \
+	} while(0)
+
+#define CHECK_DIM(w, h) \
+	do { \
+		if ((w) <= 0 || (h) <= 0 || (w) >= 0xffff || (h) >= 0xffff) { \
+			COMPLAIN_INVALID("width or height"); \
+		} \
+	} while(0)
+
 bool TextureSource::generateImagePart(std::string part_of_name,
 		video::IImage *& baseimg, std::set<std::string> &source_image_names)
 {
 	const char escape = '\\'; // same as in generateImage()
 	video::IVideoDriver *driver = RenderingEngine::get_video_driver();
 	sanity_check(driver);
 
+	if (baseimg && (baseimg->getDimension().Width == 0 ||
+		baseimg->getDimension().Height == 0)) {
```

Comment:
```suggestion
	if (baseimg && baseimg->getDimension().getArea() == 0) {
```
also "fixes" missing indent level